specifications/cloud_assembly.md

```diff
@@ -0,0 +1,398 @@
+      └───────────── The Logical ID of the Drop
+```
+
+The following escape sequences are valid:
+* `\\` encodes the `\` literal
+* `\${` encodes the `${` literal
+
+Deployment systems **SHOULD** return an error upon encountering an occurrence of the `/` literal that is not part of a
+valid escape sequence.
+
+#### Drop Type
+Every Drop has a type specifier, which allows *Cloud Assembly* consumers to know how to deploy it. The type specifiers
+are `string`s that use an URI-like syntax (`protocol://path`), providing the coordinates to a reference implementation
+for the Drop behavior.
+
+Deployment systems **MUST** support at least one protocol, and **SHOULD** support all the protocols specified in
+the following sub-sections.
+
+##### The `npm` protocol
+Type specifiers using the `npm` protocol have the following format:
+```
+npm://[@namespace/]package/ClassName[@version]
+╰┬╯    ╰────┬────╯ ╰──┬──╯ ╰───┬───╯  ╰──┬──╯
+ │          │         │        │         └─ Optional version specifier
+ │          │         │        └─────────── Fully qualified name of the Handler class
+ │          │         └──────────────────── Name of the NPM package
+ │          └────────────────────────────── Optional NPM namespace
+ └───────────────────────────────────────── NPM protocol specifier
+```
+
+#### Environment
+Environments help Deployment systems determine where to deploy a particular Drop. They are referenced by `string`s that
+use an URI-like syntax (`protocol://path`).
+
+Deployment systems **MUST** support at least one protocol, and **SHOULD** support all the protocols specified in the
+following sub-sections.
+
+##### The `aws` protocol
+Environments using the `aws` protocol have the following format:
+```
+aws://account/region
+╰┬╯   ╰──┬──╯ ╰──┬─╯
+ │       │       └─ The name of an AWS region (e.g: eu-west-1)
+ │       └───────── An AWS account ID (e.g: 123456789012)
+ └───────────────── AWS protocol specifier
+```
+
+### `Missing`
+[Drops](#drop) may require contextual information to be available in order to correctly participate in a
+*Cloud Assembly*. When information is missing, *Cloud Assembly* producers report the missing information by adding
+entries to the `missing` section of the [manifest document](#manifest-document). The values are [JSON] `object`s that
+**MUST** conform to the following schema:
+
+Key            |Type             |Required|Description
+---------------|-----------------|:------:|-----------
+`provider`     |`string`         |Required|A tag that identifies the entity that should provide the information.
+`props`        |`Map<string,any>`|Required|Properties that are required in order to obtain the missing information.
+
+### Digital Signature
+#### Signing
+*Cloud Assemblers* **SHOULD** support digital signature of *Cloud Assemblies*. When support for digital signature is
+present, *Cloud Assemblers*:
+* **MUST** require the user to specify which [PGP][RFC 4880] key should be used.
```

Comment:
Who is "the user" here?

Reply:
Again, this section feels somewhat premature to me.

Reply:
I wanna specify signature early - it's dear to my heart.